pkg/csi/service/vanilla/controller.go

```diff
@@ -712,6 +712,22 @@ func (c *controller) DeleteVolume(ctx context.Context, req *csi.DeleteVolumeRequ
 					"failed to get VolumeID from volumeMigrationService for volumePath: %q", volumePath)
 			}
 		}
+		// Check if the volume contains CNS snapshots.
+		if commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.BlockVolumeSnapshot) {
+			snapshots, err := common.QueryVolumeSnapshotsByID(ctx, c.manager.VolumeManager, req.VolumeId)
+			if err != nil {
+				return nil, logger.LogNewErrorCodef(log, codes.Internal,
```

Comment:
what is the guideline of using either `codes.Internal` or `codes.FailedPrecondition` as in L726

Reply:
I've taken that from the spec https://github.com/container-storage-interface/spec/blob/master/spec.md#deletevolume-errors

```
FAILED_PRECONDITION:
Indicates that the volume corresponding to the specified volume_id could not be deleted because it is in use by another resource or has snapshots and the plugin doesn't treat them as independent entities.
```